File: Universal Presenter Remote/Universal Presenter Remote/DBZ_LoginView.m
//
//  DBZ_LoginView.m
//  Universal Presenter Remote
//
//  Created by Brendan Boyle on 4/16/14.
//  Copyright (c) 2014 DBZ Technology. All rights reserved.
//

#import "DBZ_LoginView.h"
#import "DBZ_ServerCommunication.h"
#import <iAd/iAd.h>

#import <Google/Analytics.h>

@interface DBZ_LoginView ()

@end

@implementation DBZ_LoginView

NSTimer *refreshTimer;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.canDisplayBannerAds = YES;
    [self.navigationController.navigationBar setTitleTextAttributes: @{NSForegroundColorAttributeName: [UIColor whiteColor], NSFontAttributeName: [UIFont fontWithName:@"BatmanForeverAlternate" size:35.0f]}];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(updateInterface:) name:@"UpdateInterface" object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(refreshInterface:) name:@"Refresh" object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(openInstructions:) name:@"OpenInstructions" object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(resetQR:) name:@"ResetQR" object:nil];
    
    [DBZ_ServerCommunication setupUid];
    
    refreshTimer = [NSTimer scheduledTimerWithTimeInterval:2.5 target:self selector:@selector(localRefresh) userInfo:nil repeats:true];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)updateInterface:(NSNotification *)notification {
    int inttoken = [DBZ_ServerCommunication temptoken];
    if (inttoken > 10) {
        [_tokenLabel setText:[NSString stringWithFormat:@"%d",inttoken]];
    }
    switch ([DBZ_ServerCommunication controlmode]) {
        case 0:
            [_connectButton setEnabled:NO];
            [_connectButton setTitle:@"Connecting..." forState:UIControlStateDisabled];
            //[self updateWatchLogin:inttoken withConnectEnabled:NO withConnectText:@"Connecting..."];
            break;
        case 1:
            [_connectButton setEnabled:NO];
            [_connectButton setTitle:@"Waiting..." forState:UIControlStateDisabled];
            //[self updateWatchLogin:inttoken withConnectEnabled:NO withConnectText:@"Waiting..."];
            break;
        case 2:
            [_connectButton setEnabled:YES];
            [_connectButton setTitle:@"Begin" forState:UIControlStateNormal];
            //[self updateWatchLogin:inttoken withConnectEnabled:YES withConnectText:@"Begin"];
            break;
            
        default:
            break;
    }
}

- (void)refreshInterface:(NSNotification *)notification {
    [DBZ_ServerCommunication checkToken];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
    if ([[segue identifier] isEqualToString:@"ControlSegue"] || [[segue identifier] isEqualToString:@"SettingsSegue"])
    {
        [self.navigationController.navigationBar setTitleTextAttributes:[NSDictionary dictionaryWithObjectsAndKeys: [UIFont fontWithName:@"mplus-1c-regular" size:21],  NSFontAttributeName, nil]];
        [self.navigationController.navigationBar setTitleTextAttributes:@{NSForegroundColorAttributeName : [UIColor whiteColor]}];
    }
    
}

- (IBAction)refresh:(id)sender {
    [DBZ_ServerCommunication setupUid];
    [DBZ_ServerCommunication checkToken];
}

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    
    id<GAITracker> tracker = [[GAI sharedInstance] defaultTracker];
    [tracker set:kGAIScreenName value:@"Login"];
    [tracker send:[[GAIDictionaryBuilder createScreenView] build]];
    
    [self.navigationController.navigationBar setTitleTextAttributes: @{NSForegroundColorAttributeName: [UIColor whiteColor], NSFontAttributeName: [UIFont fontWithName:@"BatmanForeverAlternate" size:35.0f]}];
    
    [self resetQR:nil];
}

- (void)openInstructions:(NSNotification*)notification {
    [self performSegueWithIdentifier:@"InstructionSegue" sender:self];
}

- (void)resetQR:(NSNotification *)notification {
    _QRSelector.selectedSegmentIndex = 0;
}

- (void)localRefresh {
    if ([DBZ_ServerCommunication.apns isEqual: @""]) {
        // running in Simulator
        [DBZ_ServerCommunication checkToken];
    } else {
        [refreshTimer invalidate];
    }
}

@end

File: Universal Presenter Remote/Universal Presenter Remote/DBZ_LoginView.swift
//
//  DBZ_LoginView.swift
//  Universal Presenter Remote
//
//  Created by Brendan Boyle on 4/16/14.
//  Copyright (c) 2014 DBZ Technology. All rights reserved.
//

import UIKit

@objc(DBZ_LoginView)
class DBZ_LoginView: UIViewController {

    @IBOutlet weak var tokenLabel: UILabel!
    @IBOutlet weak var connectButton: UIButton!

    private var refreshTimer: Timer?

    override func viewDidLoad() {
        super.viewDidLoad()

        applyTitleFont()

        let center = NotificationCenter.default
        center.addObserver(self, selector: #selector(updateInterface(_:)), name: Notification.Name("UpdateInterface"), object: nil)
        center.addObserver(self, selector: #selector(refreshInterface(_:)), name: Notification.Name("Refresh"), object: nil)
        center.addObserver(self, selector: #selector(openInstructions(_:)), name: Notification.Name("OpenInstructions"), object: nil)

        DBZ_ServerCommunication.setupUid()

        refreshTimer = Timer.scheduledTimer(timeInterval: 2.5, target: self, selector: #selector(localRefresh), userInfo: nil, repeats: true)
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        applyTitleFont()
    }

    private func applyTitleFont() {
        var attributes: [NSAttributedString.Key: Any] = [.foregroundColor: UIColor.white]
        if let font = UIFont(name: "BatmanForeverAlternate", size: 35.0) {
            attributes[.font] = font
        }
        navigationController?.navigationBar.titleTextAttributes = attributes
    }

    @objc func updateInterface(_ notification: Notification) {
        let tempToken = DBZ_ServerCommunication.temptoken()
        if tempToken > 10 {
            tokenLabel.text = "\(tempToken)"
        }

        switch DBZ_ServerCommunication.controlmode() {
        case 0:
            connectButton.isEnabled = false
            connectButton.setTitle("Connecting...", for: .disabled)
            connectButton.backgroundColor = UIColor(named: "Disabled")
        case 1:
            connectButton.isEnabled = false
            connectButton.setTitle("Waiting...", for: .disabled)
            connectButton.backgroundColor = UIColor(named: "Disabled")
        case 2:
            connectButton.isEnabled = true
            connectButton.setTitle("Begin", for: .normal)
            connectButton.backgroundColor = UIColor(named: "Primary")
        default:
            break
        }
    }

    @objc func refreshInterface(_ notification: Notification) {
        DBZ_ServerCommunication.checkToken()
    }

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "ControlSegue" || segue.identifier == "SettingsSegue" {
            var attributes: [NSAttributedString.Key: Any] = [.foregroundColor: UIColor.white]
            if let font = UIFont(name: "mplus-1c-regular", size: 21) {
                attributes[.font] = font
            }
            navigationController?.navigationBar.titleTextAttributes = attributes
        }
    }

    @IBAction func refresh(_ sender: Any) {
        DBZ_ServerCommunication.setupUid()
        DBZ_ServerCommunication.checkToken()
    }

    @objc func openInstructions(_ notification: Notification) {
        performSegue(withIdentifier: "InstructionSegue", sender: self)
    }

    @objc func localRefresh() {
        if DBZ_ServerCommunication.apns().isEmpty {
            // running in Simulator
            DBZ_ServerCommunication.checkToken()
        } else {
            refreshTimer?.invalidate()
        }
    }
}
